apps/portal-api/src/security/directive-graphql/validators/system-token.validator.ts

```
@@ -0,0 +1,62 @@
+import config from 'config';
+import { PortalContext } from '../../../model/portal-context';
+import { SYSTEM_USER_CONTEXT } from '../../../portal.const';
+import { requestContext } from '../../../requestContext';
+import { ForbiddenAccess } from '../../../utils/error/error.util';
+
+export const SYSTEM_TOKEN_HEADER = 'x-xtm-hub-token';
+export const SYSTEM_TOKEN_VALUE = config.get<string>('system_token');
+export const SYSTEM_TOKEN_DIRECTIVE_NAME = 'system_token';
+
+export const extractSystemToken = (context: PortalContext): string | null => {
+  return (context.req?.headers?.[SYSTEM_TOKEN_HEADER] as string) || null;
+};
+
+export const isValidSystemToken = (token: string | null): boolean => {
+  if (!token) {
+    return false;
+  }
+
+  return token === SYSTEM_TOKEN_VALUE;
+};
+
+export const validateSystemToken = (context: PortalContext): boolean => {
+  const token = extractSystemToken(context);
+  const isValid = isValidSystemToken(token);
+
+  if (!isValid) {
+    throw ForbiddenAccess('Invalid system token attempt');
+  }
+
+  return isValid;
+};
+
+export const createSystemTokenResolver = (originalResolve) => {
+  return async function secureResolver(
+    source,
+    args,
+    portalContext: PortalContext,
+    info
+  ) {
+    const hasValidToken = validateSystemToken(portalContext);
+
+    if (!hasValidToken) {
+      throw new Error('Valid system token is required for this operation');
```

Do we need to throw a GraphQL error here ?

---

Yes as we call from a Graphql endpoint

---

but it's not for the front so should I add it to the enum error ?

---

To hide why it's failing ?